exporter/logicmonitorexporter/logs_exporter.go

```diff
@@ -0,0 +1,130 @@
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/logicmonitor/lm-data-sdk-go/api/logs"
+	"github.com/logicmonitor/lm-data-sdk-go/utils"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logExporter struct {
+	config          *Config
+	logger          *zap.Logger
+	logIngestClient *logs.LMLogIngest
+	settings        component.TelemetrySettings
+}
+
+const (
+	hostname         = "hostname"
+	hostnameProperty = "system.hostname"
+)
+
+// Create new logicmonitor logs exporter
+func newLogsExporter(cfg component.Config, set component.ExporterCreateSettings) *logExporter {
+	oCfg := cfg.(*Config)
+
+	return &logExporter{
+		config:   oCfg,
+		logger:   set.Logger,
+		settings: set.TelemetrySettings,
+	}
+}
+
+func (e *logExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	authParams := utils.AuthParams{
+		AccessID:    e.config.APIToken.AccessID,
+		AccessKey:   e.config.APIToken.AccessKey,
+		BearerToken: e.config.Headers["Authorization"],
+	}
+	options := []logs.Option{
+		logs.WithLogBatchingDisabled(),
+		logs.WithAuthentication(authParams),
+		logs.WithHTTPClient(client),
+		logs.WithEndpoint(e.config.Endpoint),
+	}
+
+	lli, err := logs.NewLMLogIngest(context.Background(), options...)
+	if err != nil {
+		return fmt.Errorf("failed to initialize LMLogIngest: %w", err)
+	}
+	e.logIngestClient = lli
+	return nil
+}
+
+func (e *logExporter) PushLogData(ctx context.Context, lg plog.Logs) error {
+	e.logger.Debug("Pushing log data to Logicmonitor")
+	resourceLogs := lg.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		resourceLog := resourceLogs.At(i)
+		libraryLogs := resourceLog.ScopeLogs()
+		for j := 0; j < libraryLogs.Len(); j++ {
+			libraryLog := libraryLogs.At(j)
+			logs := libraryLog.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				logMetadataMap := make(map[string]string)
+				resourceMapperMap := make(map[string]string)
+				log := logs.At(k)
+				// Copying resource attributes to log attributes
+				resourceLog.Resource().Attributes().Range(func(k string, v pcommon.Value) bool {
+					logAttributes := log.Attributes()
+					e.mergeAttributes(k, v, logAttributes)
+					return true
+				})
+				attributesMap := log.Attributes()
+				attributesMap.Range(func(key string, value pcommon.Value) bool {
+					if key == hostname {
+						resourceMapperMap[hostnameProperty] = value.Str()
+					}
+					logMetadataMap[key] = value.Str()
+					return true
+				})
+				e.logger.Debug("Sending log data", zap.String("body", log.Body().Str()), zap.Any("resourcemap", resourceMapperMap), zap.Any("metadatamap", logMetadataMap))
+				err := e.logIngestClient.SendLogs(ctx, log.Body().Str(), resourceMapperMap, logMetadataMap)
```

Comment:
From lightly looking at https://github.com/logicmonitor/lm-data-sdk-go/blob/90076d5e7403e42c3fb375b219cbebc258f4f356/api/logs/logs.go#L88 this will override the timestamp of the recording. It might make sense to use https://github.com/logicmonitor/lm-data-sdk-go/blob/90076d5e7403e42c3fb375b219cbebc258f4f356/api/logs/logs.go#L118 here and create the request with the timestamp of the log.

Reply:
Thanks for pointing it out. Updated it in SDK accordingly!